docker/optimism/optimism.sh

```diff
@@ -12,73 +11,88 @@ CACHE_DIR=$(obtain_cache_dir $0)
 DEPLOYMENT_DIR="${CACHE_DIR}/deployment"
 DEPLOYMENT_CONFIG_FILE="${CACHE_DIR}/optimism-local.json"
 KEYFILE="${CACHE_DIR}/${ADDRESS}.json"
+OPTIMISM="${CACHE_DIR}/optimism"
 
 ensure_keyfile_exists ${PRIVKEY} ${KEYFILE}
 echo Beamer deployer\'s keyfile: ${KEYFILE}
 
+configure_repo() {
+    repo="https://github.com/ethereum-optimism/optimism.git"
+    commit_id="6a474e36aba94f15ed90f71d1920e4b1d34513ab"
+    git clone --no-checkout ${repo} ${OPTIMISM}
+    cd ${OPTIMISM}
+    git checkout ${commit_id}
+}
+
+run_on_op_launcher () {
+    ex=$1
+    docker run -v /var/run/docker.sock:/var/run/docker.sock \
+        -v ${OPTIMISM}:${OPTIMISM} \
+        --network=host optimism-launcher bash -c "${ex}"
+}
+
 down() {
     echo "Shutting down the end-to-end environment"
-    make -C ${OPTIMISM}/ops down
+    run_on_op_launcher "cd ${OPTIMISM} && make devnet-down && make devnet-clean"
+}
+
+build() {
+    DOCKER_BUILDKIT=1 docker build ${ROOT}/docker/optimism -t optimism-launcher
 }
 
 up() {
     echo "Starting the end-to-end environment"
-    docker_compose_file="-f ${OPTIMISM}/ops/docker-compose.yml"
-    # FIXME: Need removal after: https://github.com/beamer-bridge/beamer/issues/1242
-    export DOCKER_TAG_MESSAGE_RELAYER=0.5.25
-    docker compose ${docker_compose_file} up --scale relayer=0 -d
-
-    echo "Wait to make sure all services are up and running"
-    # The current wait-for-sequencer.sh script in the optimism repo
-    # has a bug in that it does not specify the configuration files.
-    # So work around that here.
-    WAIT_FOR_SEQUENCER_SCRIPT=$(mktemp)-wait-for-sequencer.sh
-    sed "s#docker-compose#docker-compose ${docker_compose_file}#" \
-            "${OPTIMISM}/ops/scripts/wait-for-sequencer.sh" > ${WAIT_FOR_SEQUENCER_SCRIPT}
-    sh ${WAIT_FOR_SEQUENCER_SCRIPT}
+    if [ ! -d ${OPTIMISM} ]; then
+        configure_repo
+    fi
+    run_on_op_launcher "cd ${OPTIMISM} && make devnet-up"
```

Comment:
I think the `up` command should implicitly call `build` if there is no container image.
That way we retain the common script usage with all the other E2E test scripts.
We can still have `build` as a separate function and make it available to be invoked manually, but `up` should just call it automatically, if needed. This should also remove the need to add a separate `build` step in Optimism E2E job on CI.

Reply:
there is no build step for ci anymore